src/core/interpolation.rs

```diff
@@ -0,0 +1,243 @@
+use num_traits::Float;
+use std::ops::{Add, Mul, Sub};
+
+/// Trait for interpolation between two values
+pub trait Interpolator<T, V>
+where
+    T: Float + Copy,
+    V: Copy + Add<Output = V> + Sub<Output = V> + Mul<T, Output = V>,
+{
+    fn interpolate(&self, alpha: T, a: &V, b: &V) -> V;
+}
+
+/// Keyframes, used to construct (and define) a piecewise interpolator
+/// Generic keyframe: maps an input (query) to an output value.
+#[derive(Clone, Copy)]
+pub struct InterpolationKeyframe<T, V> {
+    pub input: T,
+    pub output: V,
+}
+
+/// Generic container for performing interpolation between keyframes
+pub struct KeyframeInterpolator<T, V, F>
```

Comment:
Not quite a fully-featured trajectory, as we're pretty limited in the interpolation types for the "segments", but sufficient for the use-cases here.